File: app/src/main/java/com/example/admin/woailiushuang/RecycleViewDemoActivity/CommonAdapter.java
package com.example.admin.woailiushuang.RecycleViewDemoActivity;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;
import java.util.zip.Inflater;

public abstract class CommonAdapter<T> extends RecyclerView.Adapter<DIYViewHolder> {
    protected Context mContext;
    protected List<T> mDatas;
    protected int mLayoutId;
    protected LayoutInflater inflater;

    public CommonAdapter(Context context, List<T> mDatas, int mLayoutId) {
        this.mContext = context;
        this.mDatas = mDatas;
        this.mLayoutId = mLayoutId;
        this.inflater = LayoutInflater.from(context);
    }

    @Override
    public DIYViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

         DIYViewHolder holder = DIYViewHolder.get(mContext,mLayoutId,parent);

        return holder;
    }

    @Override
    public void onBindViewHolder(DIYViewHolder holder, int position) {

        convert(holder,mDatas.get(position));
    }

    @Override
    public int getItemCount() {
        return mDatas.size();
    }

    public abstract void convert( DIYViewHolder holder, T t);

}

File: app/src/main/java/com/example/admin/woailiushuang/RecycleViewDemoActivity/CommonAdapter.kt
package com.example.admin.woailiushuang.RecycleViewDemoActivity

import android.content.ClipData
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup

/**
 * @desc
 *我们的每次使用过程中，针对的数据类型Bean肯定是不同的，那么这里肯定要引入泛型代表我们的Bean，内部通过一个List代表我们的数据，
 * @auth ${user}
 * @time 2019/3/15 14:09
 */
//class CommonAdapter<T>: RecyclerView.Adapter<CommonHolder> {
//    var mContext:Context?=null
//    var mData:List<T>?=null
//    var mLayoutId:Int?=null
//    var mInflater:LayoutInflater?=null
//    constructor(context: Context,layoutId:Int,data: List<T> ){
//        mLayoutId = layoutId
//        mData =data
//        mContext = context
//        mInflater = LayoutInflater.from(context)
//    }
//
//    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): CommonHolder {
//
//    }
//
//    override fun getItemCount(): Int {
//    }
//
//    override fun onBindViewHolder(p0: CommonHolder, p1: Int) {
//    }
//
//}